dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -0,0 +1,371 @@
+
+  // Getting the number of samples in the index file
+  int64_t sample_desc_num_signed;
+  DALI_ENFORCE(index_file >> sample_desc_num_signed,
+               IndexFileErrMsg(index_path, 0, "no sample count found"));
+  DALI_ENFORCE(sample_desc_num_signed > 0,
+               IndexFileErrMsg(index_path, 0, "sample count must be positive"));
+  index_file.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+  const size_t sample_desc_num = sample_desc_num_signed;
+  samples_container.reserve(samples_container.size() + sample_desc_num);
+  for (size_t sample_index = 0; sample_index < sample_desc_num; sample_index++) {
+    ParseSampleDesc(samples_container, components_container, index_file, index_path,
+                    sample_index + 1);
+  }
+}
+
+}  // namespace wds
+}  // namespace detail
+
+inline std::string SupportedTypesListGen() {
+  std::stringstream out;
+  for (auto& dtype : detail::wds::kSupportedTypes) {
+    out << dtype << ',';
+  }
+  std::string out_str = out.str();
+  return out_str.substr(0, out_str.size() - (detail::wds::kSupportedTypes.size() > 0));
+}
+
+WebdatasetLoader::WebdatasetLoader(const OpSpec& spec)
+    : Loader(spec),
+      uris_(spec.GetRepeatedArgument<std::string>("uris")),
+      index_paths_(spec.GetRepeatedArgument<std::string>("index_paths")),
+      missing_component_behavior_(detail::wds::ParseMissingExtBehavior(
+          spec.GetArgument<std::string>("missing_component_behavior"))) {
+  DALI_ENFORCE(uris_.size() == index_paths_.size(),
+               "Number of uris does not match the number of index files");
+  DALI_ENFORCE(uris_.size() > 0, "No webdataset shards provided");
+  DALI_ENFORCE(missing_component_behavior_ != detail::wds::MissingExtBehavior::Invalid,
+               "Invalid value for missing_component_behavior");
+
+  std::vector<std::string> samples_exts = spec.GetRepeatedArgument<std::string>("ext");
+  ext_.reserve(samples_exts.size());
+
+  // splitting extension bundles by the delimiter
+  for (size_t exts_idx = 0; exts_idx < samples_exts.size(); exts_idx++) {
+    std::stringstream exts_stream(samples_exts[exts_idx]);
+    std::string ext;
+    ext_.emplace_back();
+    while (std::getline(exts_stream, ext, detail::wds::kExtDelim)) {
+      if (!ext_.back().count(ext)) {
+        ext_.back().insert(ext);
+      }
+    }
+  }
+
+  auto dtypes_ids = spec.HasArgument("dtypes") ? spec.GetRepeatedArgument<DALIDataType>("dtypes") :
+                                                 std::vector<DALIDataType>(ext_.size(), DALI_UINT8);
+  dtypes_.reserve(dtypes_ids.size());
+  std::transform(dtypes_ids.begin(), dtypes_ids.end(), std::back_inserter(dtypes_),
+                 TypeTable::GetTypeInfo);
+
+  for (auto& dtype : dtypes_) {
+    DALI_ENFORCE(detail::wds::kSupportedTypes.count(dtype.id()),
+                 make_string("Unsupported output dtype ", dtype.name(),
+                             ". Supported types include: ", SupportedTypesListGen()));
```

Comment:
```suggestion
                             ". Supported types are: ", SupportedTypesListGen()));
```
I think it is a full list.

Reply:
alright